src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -896,3 +896,10 @@ def match(self, segments, parse_context):
         dialects.
         """
         return MatchResult.from_unmatched(segments)
+
+    def simple(self, parse_context: ParseContext) -> Optional[List[str]]:
+        """Does this matcher support a lowercase hash matching route?
+
+        Yes, trivially so.
+        """
+        return ["1"]
```

Comment:
I'm a bit concerned with the unintended consequences of this to be honest. But don't understand it enough.

Reply:
I went to bed Friday evening confident in my code and woke up Saturday morning with the same question.  I was hoping someone with experience in the match logic could chime in and allay or confirm my concerns.

Reply:
Maybe @alanmcruickshank or @dmateusp? I've worked on the parser, but I haven't remembered much; I have to go and review the code if I am getting into anything deep.